File: docs/experiments/code/latency/hook.go
package main

import (
	"encoding/json"
	"io"
	"os"
	"strings"
)

type ev struct {
	ToolName  string                 `json:"tool_name"`
	ToolInput map[string]interface{} `json:"tool_input"`
}

func main() {
	raw, _ := io.ReadAll(os.Stdin)
	var e ev
	if json.Unmarshal(raw, &e) != nil {
		os.Exit(0)
	}
	blob, _ := json.Marshal(e.ToolInput)
	if strings.Contains(string(blob), "secrets.env") {
		os.Stderr.WriteString("BLOCKED: secrets.env is off limits. Use config.local.json instead.")
		os.Exit(2)
	}
	os.Exit(0)
}

File: docs/experiments/code/latency/hook.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
int main(void) {
    static char buf[1 << 20];
    size_t n = fread(buf, 1, sizeof(buf) - 1, stdin);
    buf[n] = 0;
    if (strstr(buf, "secrets.env")) {
        fputs("BLOCKED: secrets.env is off limits. Use config.local.json instead.", stderr);
        return 2;
    }
    return 0;
}
